index.js

```diff
@@ -74,23 +74,25 @@ module.exports = (iterable, options) => new PCancelable((resolve, reject, onCanc
 		maxFiltered++;
 		elCount++;
 
-		Promise.resolve(el).then(
-			value => {
+		(async () => {
+			try {
+				const value = await Promise.resolve(el);
```

Comment:
`Promise.resolve` was just used to start a promise chain. It's not needed anymore.

Reply:
It is. It seems to have influence on microtask queue processing order. Without it, a test fails.

Reply:
Weird